addon/components/flash-message.js

```diff
@@ -2,127 +2,110 @@ import { htmlSafe, classify } from '@ember/string';
 import Component from '@ember/component';
 import { isPresent } from '@ember/utils';
 import { run } from '@ember/runloop';
-import { computed, set, get } from '@ember/object';
+import { action, computed, set } from '@ember/object';
+import { and, bool, readOnly, not } from '@ember/object/computed';
+import { tagName } from '@ember-decorators/component';
 import layout from '../templates/components/flash-message';
-import getWithDefault from '../utils/get-with-default';
-
-const {
-  and,
-  bool,
-  readOnly,
-  not
-} = computed;
-const {
-  next,
-  cancel
-} = run;
-
-export default Component.extend({
-  layout,
-  active: false,
-  messageStyle: 'bootstrap',
-  classNames: ['flash-message'],
-  classNameBindings: ['alertType', 'active', 'exiting'],
-  attributeBindings: ['aria-label', 'aria-describedby', 'role'],
-
-  showProgress: readOnly('flash.showProgress'),
-  notExiting: not('exiting'),
-  showProgressBar: and('showProgress', 'notExiting'),
-  exiting: readOnly('flash.exiting'),
-  hasBlock: bool('template').readOnly(),
-
-  alertType: computed('flash.type', {
-    get() {
-      const flashType = getWithDefault(this, 'flash.type', '');
-      const messageStyle = getWithDefault(this, 'messageStyle', '');
-      let prefix = 'alert alert-';
-
-      if (messageStyle === 'foundation') {
-        prefix = 'alert-box ';
-      }
-
-      return `${prefix}${flashType}`;
-    }
-  }),
 
-  flashType: computed('flash.type', {
-    get() {
-      const flashType = getWithDefault(this, 'flash.type', '');
+const { next, cancel } = run;
 
-      return classify(flashType);
-    }
-  }),
+@tagName('')
```

Comment:
Is this just to make it easier to transition to glimmer components? Might be a breaking change for some consumers with little benefit.

Reply:
Yeah, you're probably right, I didn't run try before I opened the PR so I didn't realize how many older versions broke. I would attempt to move this work to the 2.0 branch, but that hasn't been touched in four years. I was attempting to breathe some life into this project, but seems like without the help of the maintainers it will be an uphill battle. I'll close this PR.

Reply:
Sorry, I think this PR is 99% of the way there! Nearly all these changes are net wins. The only one I was worried about was this tagName decorator.

Reply:
Ah, I suppose I could leave this PR open and leave it up to the maintainers to determine if they're ok breaking older versions, I just assumed that it would be a deal-breaker.

As for your concerns about making this tagless, I'll point out that I added a wrapping div to the template that implements all the bindings we had before, namely classnames and applying various attributes, which is now handled via `...attributes`.

That said, this would introduce a change in markup in apps that do provide a custom tag type to the component and could potentially break them. I think going forward this approach is more in line with where we're heading with glimmer components, but I can remove this change if you think it's going to break consumers.

Reply:
I don't have any data to back it up, but I have a hunch that customizing the tagName is pretty rare. Adding the div in the template satisfies my main concern which was breaking styles that relied on that HTML structure.

Reply:
Yeah, that was my thought too. If a user does use a custom tag it would change their markup only slightly, but it should be an easy fix (famous last words).